packages/excalidraw/actions/actionProperties.tsx

```diff
@@ -1571,7 +1571,8 @@ export const actionChangeArrowType = register({
         if (isElbowArrow(newElement)) {
           const elementsMap = app.scene.getNonDeletedElementsMap();
 
-          app.dismissLinearEditor();
+          //defer so that previous changes don't get overwritten
+          setTimeout(() => app.dismissLinearEditor());
```

Comment:
@mtolmacs missed this previously, but I'm assuming this is only needed for simple->elbow type change? If so, I'd make it conditional so we don't exit the editor when changing between round and sharp simple arrow types.

Also, now that it works "correctly", it unearthed a bug where the elbow arrows seems deselected on type change because the points are not rendered at all.

Reply:
This part was working the same previously, the only difference was that the `setTimeout` was being set in the `app.dismissLinearEditor` function.

Reply:
@dwelle it is conditional, two lines above the `if (isElbowArrow(newElement)) {` line limits this effect to conversion into elbow arrow and it does not affect changes from sharp to curved or vice versa.

Unfortunately the change in actionFinalize.ts (`selectionElements: null`) dismisses the selection, which is the reason for the regression you experience with converting a simple arrow with active line editor into an elbow arrow. In master the elbow arrow remains selected.

Reply:
Sorry, the `selectedLinearElement: null` in actionFinalize.ts causes the loss of selection, not `selectionElement: null`.